editoast/src/views/mod.rs

```diff
@@ -105,14 +109,23 @@ async fn health(
     db_pool: Data<DbConnectionPoolV2>,
     redis_client: Data<RedisClient>,
 ) -> Result<&'static str> {
-    use diesel_async::RunQueryDsl;
-    sql_query("SELECT 1")
-        .execute(db_pool.get().await?.deref_mut())
-        .await?;
+    timeout(
+        Duration::from_millis(500),
+        check_health(db_pool.into_inner(), redis_client.into_inner()),
+    )
+    .await
+    .map_err(|_| EditoastModelsError::TimeoutError)??;
+    Ok("ok")
+}
 
+async fn check_health(
+    db_pool: Arc<DbConnectionPoolV2>,
+    redis_client: Arc<RedisClient>,
+) -> Result<()> {
+    ping_database(db_pool.get().await?.deref_mut()).await?;
     let mut conn = redis_client.get_connection().await?;
-    cmd("PING").query_async::<_, ()>(&mut conn).await.unwrap();
-    Ok("ok")
+    cmd("PING").query_async::<_, ()>(&mut conn).await?;
```

Comment:
this could probably have the same timeout and be run at the same time as the ping